File: app/src/main/java/com/asfoundation/wallet/billing/analytics/EventSender.java
package com.asfoundation.wallet.billing.analytics;

public interface EventSender {

  void sendPurchaseDetailsEvent(String packageName, String skuDetails, String value,
      String transactionType);

  void sendPaymentMethodDetailsEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType);

  void sendActionPaymentMethodDetailsActionEvent(String packageName, String skuDetails,
      String value, String purchaseDetails, String transactionType, String action);

  void sendPaymentEvent(String packageName, String skuDetails, String value, String purchaseDetails,
      String transactionType);

  void sendRevenueEvent(String value);

  void sendPreSelectedPaymentMethodEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String action);

  void sendPaymentMethodEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String action);

  void sendPaymentConfirmationEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String action);

  void sendPaymentErrorEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String errorCode);

  void sendPaymentErrorWithDetailsEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String errorCode, String errorDetails);

  void sendPaymentErrorWithDetailsAndRiskEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String errorCode, String errorDetails,
      String riskRules);

  void sendPaymentSuccessEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType);

  void sendPaymentPendingEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType);

  void sendPurchaseStartEvent(String packageName, String skuDetails, String value,
      String purchaseDetails, String transactionType, String context);

  void sendPurchaseStartWithoutDetailsEvent(String packageName, String skuDetails, String value,
      String transactionType, String context);

  void sendPaypalUrlEvent(String packageName, String skuDetails, String value,
      String transactionType, String type, String resultCode, String url);
}

File: core/analytics/src/main/java/com/appcoins/wallet/core/analytics/analytics/legacy/EventSender.kt
package com.appcoins.wallet.core.analytics.analytics.legacy

interface EventSender {
  fun sendPurchaseDetailsEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    transactionType: String?,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentMethodDetailsEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendActionPaymentMethodDetailsActionEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    action: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendRevenueEvent(value: String, isOnboardingPayment: Boolean = false)

  fun sendPreSelectedPaymentMethodEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String?,
    action: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentMethodEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String?,
    action: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentConfirmationEvent(
    packageName: String?,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String?,
    action: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentErrorEvent(
    packageName: String,
    skuDetails: String,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    errorCode: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentErrorWithDetailsEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    errorCode: String,
    errorDetails: String?,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentErrorWithDetailsAndRiskEvent(
    packageName: String,
    skuDetails: String,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    errorCode: String,
    errorDetails: String?,
    riskRules: String?,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaymentSuccessEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    isOnboardingPayment: Boolean = false,
    txId: String,
    valueUsd: String
  )

  fun sendPaymentPendingEvent(
    packageName: String,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPurchaseStartEvent(
    packageName: String?,
    skuDetails: String?,
    value: String,
    purchaseDetails: String,
    transactionType: String?,
    context: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPurchaseStartWithoutDetailsEvent(
    packageName: String?,
    skuDetails: String?,
    value: String,
    transactionType: String?,
    context: String,
    isOnboardingPayment: Boolean = false
  )

  fun sendPaypalUrlEvent(
    packageName: String?,
    skuDetails: String?,
    value: String,
    transactionType: String,
    type: String?,
    resultCode: String?,
    url: String?,
    isOnboardingPayment: Boolean = false
  )
}